### PR TITLE
feat(plugin): allow some one-line arrow functions

### PR DESCRIPTION
**Pull request by zanminkian**

### CODE REVIEW

File: packages/eslint-plugin-antfu/src/rules/top-level-function.test.ts [hunk 1]
```diff
@@ -17,15 +17,39 @@ const valids = [
...
+  'const foo = () => String(123)',
+  'const foo = async (x, y) => x + y',
+  'const foo = async (x) => `x: ${x}`',
+  'const foo = async () => \'hello\'',
+  'const foo = async () => 123',
+  'const foo = async () => false',
+  'const foo = async () => /abcd/',
+  'const foo = async () => undefined',
+  'const foo = async () => null',
+  'const foo = async () => String(123)',
```
Comment (antfu) [inline packages/eslint-plugin-antfu/src/rules/top-level-function.test.ts:38]: It doesn't need to be this much I guess. 😅

File: packages/eslint-plugin-antfu/src/rules/top-level-function.ts [hunk 2]
```diff
@@ -40,6 +40,17 @@ export default createEslintRule<Options, MessageIds>({
...
         if (declaration.id.typeAnnotation)
           return
+        if (
+          [
+            'BinaryExpression',
+            'Literal',
+            'TemplateLiteral',
+            'Identifier',
+            'CallExpression',
+          ].includes(declaration.init?.body.type)
```
Comment (antfu) [inline packages/eslint-plugin-antfu/src/rules/top-level-function.ts:50]: I don't actually mind bypassing any one-liner

File: packages/eslint-plugin-antfu/src/rules/top-level-function.ts [hunk 3]
```diff
@@ -40,6 +40,11 @@ export default createEslintRule<Options, MessageIds>({
           return
         if (declaration.id.typeAnnotation)
           return
+        if (
+          declaration.init.body.type !== 'BlockStatement'
```
Comment (zanminkian) [inline packages/eslint-plugin-antfu/src/rules/top-level-function.ts:44]:
```js
const fun = () => { return 'abc' }
```
Code above (block statement) will be formatted.

**Comment by antfu** [comment]
Exactly! Thanks

**Pull request merged**